sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveClientSuites.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.client
+
+import scala.collection.immutable.IndexedSeq
+
+import org.scalatest.Suite
+
+class HiveClientSuites extends Suite with HiveClientVersions {
+  override def nestedSuites: IndexedSeq[Suite] = {
+    // Hive 0.12 does not provide the partition filtering API we call
```

Comment:
No test coverage has been lost. `HiveClientSuite` is not a good name for that class. It actually just tests partition filtering. I added a `TODO` suggesting we rename it to `HivePartitionFilteringSuite`, but that makes the diff difficult to grok. All other Hive client testing is still in `VersionsSuite`, and that suite still tests all supported Hive versions. I wanted to change the names of these tests to make them clearer, but it resulted in a jumbled diff. You requested that I keep the names as-is to make the diff clearer.

I added "TODO" comments suggesting better names for these classes, for renaming later.